matlab/src/cpp/arrow/matlab/error/error.h

```diff
@@ -0,0 +1,41 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+
+#include "arrow/status.h"
+#include "libmexclass/error/Error.h"
+
+#include <string_view>
+
+#define MATLAB_ERROR_IF_NOT_OK(expr, id)                               \
+    do {                                                               \
+        arrow::Status _status = (expr);                                \
+        if (!_status.ok()) {                                           \
+            return libmexclass::error::Error{(id), _status.message()}; \
+        }                                                              \
+    } while (0)
+
+namespace arrow::matlab::error {
+    // TODO: Make Error ID Enum class to avoid defining static constexpr
+    static const std::string APPEND_VALUES_ERROR_ID = "arrow:matlab:proxy:make:FailedToAppendValues";
```

Comment:
Is this `static const std::string` safe in C++?
I think that we need to use `static const char*` or `static constexpr std::string_view` instead of `static const std::string`.

Reply:
That's a good point. I'll make it a `static const char*`